### ISSUE TITLE
Trigger decorator on undo and redo

### ISSUE THREAD

**Issue by babakfp** [moalamri/vscode-inline-fold#98]
## Color preview

![Screenshot 2023-02-01 101612](https://user-images.githubusercontent.com/44144724/216129174-e38a88e8-30e3-4395-b2a1-9dc3e9de27f7.png)

![Screenshot 2023-02-01 101650](https://user-images.githubusercontent.com/44144724/216128971-5618f1fa-6549-404f-ab13-f33beb32dd14.png)

## Regex

![Screenshot 2023-02-01 101723](https://user-images.githubusercontent.com/44144724/216128980-7b573801-04ee-4325-bf4a-65e1d42c9ba4.png)

![Screenshot 2023-02-01 101739](https://user-images.githubusercontent.com/44144724/216128991-961db8c8-19cb-4353-9df1-c0dd2cb861eb.png)

**Comment by e-medeiros** [comment]
@babakfp could you provide further information?

**Comment by babakfp** [comment]
## Color preview

This would be the expected result:

![Screenshot 2023-02-01 132250](https://user-images.githubusercontent.com/44144724/216166310-81c42867-eb75-4e31-b914-5b5017742b70.png)

I would like to keep the color preview feature and hide it when classes are collapsed. (I don't know if the color preview feature is added by an extension or vscode itself).

Also, see this:

![Screenshot 2023-02-01 133252](https://user-images.githubusercontent.com/44144724/216168343-220ac724-7b84-4af8-bfc1-cdaa82793845.png)

![Screenshot 2023-02-01 133302](https://user-images.githubusercontent.com/44144724/216168517-e18db019-a139-4507-8581-571ab09787a8.png)

If a class with a color preview is the first or the only class, then its preview stays visible.


## Regex

This would be the expected result:

![Screenshot 2023-02-01 132913](https://user-images.githubusercontent.com/44144724/216167477-2e5f9dc3-8e41-4adb-bb9f-d7d2af89ad0b.png)

It seems like it doesn't play well with curly brackets wrapped inside quotation marks.

**Comment by e-medeiros** [comment]
Could you provide also...

**Comment by babakfp** [comment]
Sorry, what do you want me to provide?

**Comment by moalamri** [comment]
@e-medeiros I could be wrong, but I think it's VSCode itself.

@babakfp There's an option in VSCode settings to disable the color box `"editor.colorDecorators": false`
There are several extensions that could add colors to code by different methods hopefully in a way that will not conflict with Inline Fold extension.

**Comment by babakfp** [comment]
> @e-medeiros I could be wrong, but I think it's VSCode itself.
> 
> @babakfp There's an option in VSCode settings to disable the color box `"editor.colorDecorators": false` There are several extensions that could add colors to code by different methods hopefully in a way that will not conflict with Inline Fold extension.

I know, but I like it and I don't want to disable it.

---

I removed the Inline Fold extension because it was causing issues when I was doing `ctrl+z` and `ctrl+shift+z`. See this video: https://uploadb.me/zz7wp7068r7j.html
When the cursor line goes inside `...` it doesn't expand/show the classes.

**Comment by moalamri** [comment]
That is strange, did you have the ` inlineFold.unfoldOnLineSelect` set to true?

**Comment by babakfp** [comment]
Yes, this is my config:
```
	"inlineFold.maskColor": "#F6C177",
	"inlineFold.unfoldedOpacity": 1,
	"inlineFold.unfoldOnLineSelect": true,
```

**Comment by babakfp** [comment]
> When that option is set to true it will unfold the content whenever the cursor is on the same line. `Ctrl + z` would take you back to the same line you were editing.
> 
> Regarding the color box, there's no way I could think of that the extension could do to override it. I hope VS Code could inspect if any decoration is done to specific range then it could disable the builtin editor's helpers so it doesn't conflict with extensions performing custom decorations. For now, there's no other option than disabling the editor's color box.
> 
> I will close this issue, feel free to reopen if needed

Yes, but it doesn't work when the cursor goes inside the collapsed part. You can see it in this video: https://uploadb.me/zz7wp7068r7j.html

@moalamri

**Comment by moalamri** [comment]
@babakfp the video you shared didn't load for me and it kept timing-out, would you please share it somewhere else maybe YouTube? I'm curious to see the behavior in action, perhaps we might need to add an extra trigger maybe on `undo`, `redo`.

**Comment by babakfp** [comment]
Sorry about that. I wasn't able to upload it on github, but now here it is:

https://user-images.githubusercontent.com/44144724/216731986-0abecea8-85a6-4776-a7da-9e73d21c4b7a.mp4

At first, you can see the cursor goes from left to right. After finishing the line, I press `ctrl+z` and the cursor moves to its previous positions. Later I press `ctrl+shift+z`... so far everything works correctly. Layer I add a new class, I move the cursor inside quotes (and after writing the new class name, I move the cursor down with arrow keys). After that, I do `ctrl+z` and `ctrl+shift+z` and can see the cursor diaspores because the cursor moved to its' previous position (inside quotes) while the classes were collapsed.

**Comment by moalamri** [comment]
Thanks for reuploading the video, we must register an extra event that gets triggered when the editor's text changes. I will have a look at this later, I might change the issue title for better clarification.

**Comment by moalamri** [comment]
Fixed in v0.2.3
